### DISCUSSION TITLE
Installing Slither

### DISCUSSION THREAD

**Discussion by john-lawniczak** [Cyfrin/foundry-full-course-cu#409] [Q&A] [answered]
Hi all! 

I am trying to install Slither on my Mac using zsh.

I have installed slither using `pip3 install slither-analyzer`. However, per this [stack exchange](https://ethereum.stackexchange.com/questions/119871/slither-command-not-found): the path where pip3 stored slither, `$HOME/.local/bin/` isn't in my $PATH environment variable.

The .zshrc file also suggests a path coming from bash.
```
# If you come from bash you might have to change your $PATH.
# export PATH=$HOME/bin:/usr/local/bin:$PATH

# Path to your oh-my-zsh installation.
export ZSH="$HOME/.oh-my-zsh"
```
How can I change the path to install slither correctly?

I know it's an easy path fix... any help is appreciated.

A similar issue can be found [here](https://github.com/crytic/slither/issues/1264)

**Comment by ChefAharoni** [comment]
Try ChatGPT's answer:

To add `$HOME/.local/bin to your $PATH` environment variable in zsh, you need to modify your `.zshrc` file. Here's how to do that:

1. Open your terminal.
2. Open the .zshrc file in a text editor. For example, you could use nano:
```bash
nano ~/.zshrc
```
3. Add the following line to the file:
```bash
export PATH=$HOME/.local/bin:$PATH
```
This line of code is adding $HOME/.local/bin to your $PATH. The $PATH variable is a list of directories that your shell searches through when you enter a command.
4. Save your changes and close the file. If you're using nano, you can do this by pressing `Ctrl+O` to save the file, then `Ctrl+X` to close it.
5. To make sure your changes take effect, you need to restart your terminal session, or you can source your `.zshrc` file with the following command:
```bash
source ~/.zshrc
```
6. Now, you can check if the path was correctly added by executing:
```bash
echo $PATH
```

You should see `$HOME/.local/bin` somewhere in the output.
Now, you should be able to run slither from the terminal.

Remember, when editing your `.zshrc` file, you need to be careful. Making mistakes in this file can cause problems with your terminal. Always make sure to back up your `.zshrc` file before making changes.

**Reply by john-lawniczak** [reply]
Thanks for the reply. I am still having issues with the path, but I will continue to work on it and report back!

**Reply by john-lawniczak** [reply, marked as answer]
An update:
 
It took a long time but I think this [stack-overflow](https://stackoverflow.com/questions/42870537/zsh-command-cannot-found-pip) helped resolve the issue. To clarify I am using `.oh-my-zsh` and `homebrew`. 

In the ~ directory, input the command:
```
nano .bash_profile
```
You should see codes like:
```
# Setting PATH for Python 3.5
# The original version is saved in .bash_profile.pysave
PATH="/Library/Frameworks/Python.framework/Versions/3.5/bin:${PATH}"
export PATH
```
Copy and paste them to the end of your .zshrc file, save and exit. 
Reopen the terminal and don't forget to `source ~/.zshrc` and pip can be used to install Slither.

Sometimes setting up the local environment can be the biggest hurdle. But it's worth it for such an incredible tool like Slither.

Hope this helps someone if they encounter the same problem!

**Comment by ezesundayeze** [comment]
For reference purpose in case someone gets here looking for a solution. You can do `brew install slither-analyzer` to install slither on your mac

**Reply by EngrPips** [reply]
Interesting, thanks for the heads-up.
